tdms/src/argument_parser.cpp

```diff
@@ -0,0 +1,120 @@
+#include "stdexcept"
+#include "argument_parser.h"
+
+using namespace std;
+
+
+ArgumentParser::ArgumentParser() = default;
+
+ArgumentNamespace ArgumentParser::parse_args(int nargs, char *argv[]) {
+
+  auto args = ArgumentNamespace(nargs, argv);
+
+  if (args.have_flag("-h")){
+    print_help_message();
+    exit(0);
+  }
+
+  if( (args.num_non_flag != 2) &&  (args.num_non_flag != 3) ){
+    fprintf(stderr,"incorrect number of arguments\n");
```

Comment:
```suggestion
    fprintf(stderr,"incorrect number of arguments.\n\n");
    print_help_message();
```
It's always helpful to tell the user what to do to fix - I think the help message would work well here for that.